.github/workflows/test-flux.yaml

```diff
@@ -0,0 +1,42 @@
+name: Test Flux Executor
+on:
+  pull_request: []
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      packages: read
+    strategy:
+      fail-fast: false
+      matrix:
+        container: ['fluxrm/flux-sched:focal']
+
+    container:
+      image: ${{ matrix.container }}
+      options: "--platform=linux/amd64 --user root -it"
+
+    name: ${{ matrix.container }}
+    steps:
+      - name: Make Space
+        run: |        
+          rm -rf /usr/share/dotnet
+          rm -rf /opt/ghc
+
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install Snakemake
+        run: |
+          wget --quiet https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O ~/miniconda.sh           
+          /bin/bash ~/miniconda.sh -b -p /opt/conda
+          export PATH=/opt/conda:/bin:$PATH
+          # Install snakemake
+          pip install .
+
+      - name: Start Flux and Test Workflow
+        run: |
+          su fluxuser
+          cd examples/flux
+          which snakemake
+          flux start snakemake --flux --jobs=1
```

Comment:
Can you explain this? I would have expected that flux is started with `flux start`, and then snakemake is started with `snakemake --flux` to submit jobs to flux.

Reply:
If you do `flux start` on it's own, it essentially opens up an interactive shell (and then you can go as you please to interact with flux). To run in actions (or to do in one command) if you provide a script (or other binary to run) to flux start, it runs that (in the context of flux) so it will function as expected. If we do flux start (newline) followed by snakemake run in actions, I suspect it will hang when it hits the interactive shell, or GitHub actions will kill the shell and then the snakemake command will fail.

But to be clear, this is different from when you interact with flux on a cluster where it is installed - you typically don't need to run flux start because you already have access. The exception would be if you have, for example, a SLURM cluster and you install flux on your own to use under an allocation, you would again be calling flux start. And if you wanted something to run programatically, you'd probably do `flux start <command>` as we do here.

Conceptually (and I might not be the best one to explain this) Flux start is creating a flux instance https://flux-framework.readthedocs.io/projects/flux-core/en/latest/man1/flux-start.html with one or more brokers. I can do this on my own as a user that installed my own Flux (or in a container I just created, akin to the action) or there can already be a resource with brokers running I have access to. I think in the future we could have support for some kind of background flux start, but for now this is the strategy I've been using in CI setups running flux.

Reply:
Ok, but still. To me, this one: https://flux-framework.readthedocs.io/en/latest/quickstart.html#starting-a-flux-instance
looks like we could as well do
```
flux start --test-size=4
snakemake --flux --jobs 1
```
in order to have a mode of execution that is closer to reality. Or am I still not understanding this correctly?

Reply:
That command starts an interactive shell - it will work interactively but not in the CI.

Reply:
Would you like me to try it out just for a sanity check and to satisfy your curiosity? Happy to do that - and hey maybe I'm wrong!

Reply:
Here you go: https://github.com/snakemake/snakemake/actions/runs/3177693238/jobs/5178416860 let me know when you've seen it and I'll change it back.

Reply:
```
flux start --test-size=4
snakemake --flux --jobs 1
```

`flux start --test-size=N` starts a test flux instance with N brokers and sets up an environment such that that instance can be used and then spawns a command (by default an interactive shell) with that environment (e.g. like `/usr/bin/env`). Without running _under_ `flux start`, the subsequent `snakemake` would have no way to connect to Flux.

It works this way so we can run many tests in parallel, each under its own `flux start`.

Reply:
Ok, understood. I think that might need some clarification in your docs. Maybe I was also just blind.